lib/Conversion/TorchToLinalg/TorchToLinalg.cpp

```diff
@@ -909,30 +1003,414 @@ class ConvertAtenNativeLayerNormOp
         rewriter
             .create<linalg::GenericOp>(
                 loc, initLayerNormTensor.getType(),
-                ValueRange{input, mean, var, weight, bias}, initLayerNormTensor,
+                ValueRange{input, mean, invSTD, weight, bias},
+                initLayerNormTensor,
                 /*indexingMaps=*/indexingMaps,
                 /*iteratorTypes=*/layerNormIterationTypes,
                 [&](OpBuilder &b, Location loc, ValueRange args) {
-                  Value input = args[0], mean = args[1], var = args[2],
+                  Value input = args[0], mean = args[1], invSTD = args[2],
                         weight = args[3], bias = args[4];
                   Value result = createLinalgPayloadCalculationForNormOps(
-                      b, loc, elemTy, input, mean, var, eps, weight, bias);
+                      b, loc, elemTy, input, mean, invSTD, eps, weight, bias);
                   b.create<linalg::YieldOp>(loc, result);
                 })
             .getResult(0);
+    SmallVector<bool> meanVarBroadcastMask(inputRank, false);
+    for (unsigned i = 0; i < inputRank; i++) {
+      if (i >= meanAndVarShapeRank)
+        meanVarBroadcastMask[i] = true;
```

Comment:
Can the calculation of `broadcastMask` be done inside the `broadcast` function? It will make it more usable and modular.

Reply:
We don't need `SmallVector<bool> meanVarBroadcastMask`. We could just use `meanAndVarShapeRank` for all the places using `broadcastMask` including the uses inside the `broadcast` function.